server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -1158,6 +1160,63 @@ func (s *BuildBuddyServer) GetEventLogChunk(ctx context.Context, req *elpb.GetEv
 	return resp, err
 }
 
+func (s *BuildBuddyServer) GetEventLog(req *elpb.GetEventLogChunkRequest, stream bbspb.BuildBuddyService_GetEventLogServer) error {
+	ctx := stream.Context()
+	// Fetch the event log once as soon as we get the request, once whenever
+	// we see an update from Redis, and once every 3s (as a fallback).
```

Comment:
if we're still willing to do every 3s, is it really worth tracking in redis?  if we don't track anything in redis, we're only going to do extra work when someone is actually looking at the invocation vs. the pubsub approach, which writes to redis no matter what, right?

Reply:
Not sure I fully got your question, but these Redis pubsub writes/reads should be extremely cheap (relative to remote execution, and also relative to the existing invocation log tail that is stored in Redis)

The 3s poll is just a fallback in case Redis is unavailable for any reason (e.g. it restarts, or for OSS instances). It'll also let us get rid of the other log streaming endpoint in favor of just this one RPC.

Reply:
Sorry, all I was really asking here was why bother doing pubsub if you're going to do a 3s fallback anyway?

Reply:
I think I answered your question but just in case :sweat_smile: 
* The "happy" path is that this is a BB cloud instance and Redis is 100% available - in this case we'll only be reading from the pubsub chan in practice, since bazel sends updates much more frequently than once per 3s
* The "sad" path is either this is a self-hosted BB instance, or this is a cloud instance and redis restarted or something. In this case we do the 3s fallback so that we're updating the logs at least as frequently as we do today